src/geo.c

```diff
@@ -43,6 +43,10 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "geo_ymfm.h"
 #include "geo_z80.h"
 
+#ifdef __LIBRETRO__
```

Comment:
Why is this necessary?

Reply:
Because adding `file_stream_transforms.h` causes `stdio` functions like fopen to be mapped to retroarch's VFS implementation. If you are building the core standalone this would cause the build to fail.

Reply:
This is not something I can get on board with. Basically, there should be no #ifdefs added for any specific platform anywhere in the `src/` directory. If there is a case where something inside here breaks, then it's a case where a generic implementation can replace whatever breaks, and any platform-specific code can piggyback on that from the outside.

If you take a look at `geo_mem_ptr` it gives you the ability to manipulate any raw memory block in the emulator. This can be used to build libretro VFS implementations in libretro.c. I believe the only places this would be necessary would be for memory cards, NVRAM, and (rarely, just jockeygp and vliner) cartridge RAM aka SRAM. If this is not enough, then I can expose whatever else is necessary to do everything in libretro.c.